shared/types/Tournament.d.ts

```diff
@@ -1,26 +1,35 @@
 import User from "./User";
-
+ 
 export interface Team {
-  _id?: string;
+  id?: string;
   name: string;
-  members: User[];
+  members: Array<User>; 
 }
 
 export interface Match {
-  id: string;
-  rivals: {
-    teamA: Team;
-    teamB: Team;
-  };
-  score: {
-    teamA: number | null;
-    teamB: number | null;
-  };
+  id?: string;
+  sideA: {
+    team: Team | string; 
+    score: {
+      a: number;
+      b: number;  
+    },
+  },
+  sideB: {
+    team: Team | string;  
+    score: {
+      a: number;
+      b: number;  
+    },
+  },
+  isFinished: boolean;
+  date: Date;
 }
 
 export default interface Tournament {
+  id?: string;
   name: string;
-  owner: User;
-  participants: Team[];
-  matches: Match[];
+  ownerMId: string;
+  teams:  Array<Team | string>; 
```

Comment:
```suggestion
  teams:  Array<Team>; 
```
Same remark as above